test/controllers/TaxAccountSummaryControllerSpec.scala

```diff
@@ -63,9 +64,22 @@ class TaxAccountSummaryControllerSpec extends PlaySpec
 
     "display the income tax summary page" in {
       val sut = createSUT
-      when(employmentService.employments(any(), any())(any())).thenReturn(Future.successful(Seq(employment)))
-      when(taxAccountService.taxCodeIncomes(any(), any())(any())).thenReturn(
-        Future.successful(TaiSuccessResponseWithPayload[Seq[TaxCodeIncome]](taxCodeIncomes)))
+      when(employmentService.ceasedEmployments(any(), any())(any())).thenReturn(
+        Future.successful(Seq(employment))
+      )
+
+      when(taxAccountService.incomeSources(any[Nino], any[TaxYear], Matchers.eq("PensionIncome"), Matchers.eq("Live"))(any[HeaderCarrier])).thenReturn(
+        Future.successful(TaiSuccessResponseWithPayload[Seq[IncomeSource]](Seq.empty[IncomeSource]))
+      )
+
+      when(taxAccountService.incomeSources(any[Nino], any[TaxYear], Matchers.eq("EmploymentIncome"), Matchers.eq("Live"))(any[HeaderCarrier])).thenReturn(
+        Future.successful(TaiSuccessResponseWithPayload[Seq[IncomeSource]](Seq.empty[IncomeSource]))
+      )
+
+      when(taxAccountService.incomeSources(any[Nino], any[TaxYear], Matchers.eq("EmploymentIncome"), Matchers.eq("Ceased"))(any[HeaderCarrier])).thenReturn(
+        Future.successful(TaiSuccessResponseWithPayload[Seq[IncomeSource]](Seq.empty[IncomeSource]))
+      )
+
```

Comment:
What happens when these are not empty? Does your expectation change if any of the request fail? I think moving the function  

 private def taxAccountSummaryViewModel(nino: Nino, taxAccountSummary: TaxAccountSummary)

to a service will allow you to test these "edge" cases in a clean way, separating the concerns of building the ViewModel and rendering the ViewModel

Reply:
moved to new service